Add a free-form SETUP parameter identifying the implementation (#1114)
moq-wg/moq-transport: draft-ietf-moq-transport.md

of this kind by purging any Token Aliases that failed to register based on the
MAX_AUTH_TOKEN_CACHE_SIZE parameter in SERVER_SETUP (or the default value of 0).

> #### MOQT AGENT

Reviewer: the word Agent is bit overloaded these days.  I am fine with what you have, but make it explcit as User-Agent felt better. either works

> #### MOQT AGENT

Reviewer 2: Calling both ends user-agent is going to be super confusing.

Agent is overloaded I agree.

How about something like "IMPLEMENTATION_HINT", since this is really an optional hint containing unstructured data about implementation details

> #### MOQT AGENT

Author: Dang Agent is overloaded.  I agree "USER" doesn't belong in the name here - it may be in use when no users are involved.  I like IMPLEMENTATION, but don't love HINT.  True this is completely unauthenticated and people can and will lie.  Let me try MOQT_IMPLEMENTATION.

#bikeshed

> The MOQT_IMPLEMENTATION parameter (Parameter Type 0x05) is a UTF-8 encoded

Reviewer 3: Unless you are going to provide all the string prep rules and lang stuff for this string, I would not use UTF-8. It just bring useless review for something that is not meant to be human readable.

Author: For reference that is:

> Additional diagnostic information for the closure. This can be zero length if the sender chooses not to give details beyond the Error Code value. This SHOULD be a UTF-8 encoded string [[RFC3629](https://datatracker.ietf.org/doc/html/rfc3629)], though the frame does not carry information, such as language tags, that would aid comprehension by any entity other than the one that created the text.

> #### MOQT AGENT
> 
> The MOQT_AGENT parameter (Parameter Type 0x05) is a UTF-8 encoded
> string describing the software agent of the sender.

Author: ```suggestion
#### MOQT IMPLEMENTATION

The MOQT_IMPLEMENTATION parameter (Parameter Type 0x05) is a UTF-8 encoded
string describing the software agent of the sender.
```

> The MOQT_IMPLEMENTATION parameter (Parameter Type 0x05) is a UTF-8 encoded
> string describing the sender's MOQT implementation.

Author: ```suggestion
The MOQT_IMPLEMENTATION parameter (Parameter Type 0x05) identifies the name and
version of the sender's MOQT implementation.  This SHOULD be a UTF-8 encoded
string [RFC3629], though the message does not carry information, such as
language tags, that would aid comprehension by any entity other than the one
that created the text.
```

exhaustion attacks by a peer that does not send expected data within
an expected time.  Each implementation is expected to set its own limits.

> TODO: Security Considerations of MOQT_IMPLEMENTATION parameter

Author: ```suggestion
TODO: Security/Privacy Considerations of MOQT_IMPLEMENTATION parameter
```
